src/__tests__/integration-test.ts

```diff
@@ -2,11 +2,81 @@ import * as express from "express";
 import * as request from "supertest";
 import { ApolloServer } from "apollo-server-express";
 import ApolloOpentracing from "../";
-import * as bodyParser from "body-parser";
-import { makeExecutableSchema } from "graphql-tools";
+
+let mockSpanId = 1;
+
+interface MockSpan {
+  id: number;
+  parentId?: number;
+  name: string;
+  options?: any;
+  logs: any[];
+  finished: boolean;
+}
+
+interface MockSpanTree extends MockSpan {
+  children: MockSpanTree[];
+}
+
+const buildSpanTree = (spans: MockSpan[]) => {
+  // TODO we currently assume there is only one null parent entry.
+  // The root span
+
+  let rootSpan = null;
+
+  const spansByParentId = spans.reduce((acc, span) => {
+    // Check for root
+    if (span.parentId) {
+      if (acc.has(span.parentId)) {
+        acc.get(span.parentId).push(span);
+      } else {
+        acc.set(span.parentId, [span]);
+      }
+    } else {
+      rootSpan = span;
+    }
+
+    return acc;
+  }, new Map<number, MockSpan[]>());
+
+  expect(rootSpan).toBeDefined();
+
+  const tree = {
+    ...rootSpan,
+    children: []
+  };
+
+  buildTree(tree, spansByParentId);
+
+  // Lost Spans
+  expect(spansByParentId.size).toBe(0);
+
+  return tree;
+};
+
+const buildTree = (
+  parent: MockSpanTree,
+  spansByParentId: Map<number, MockSpan[]>
+) => {
+  if (spansByParentId.has(parent.id)) {
+    const spans = spansByParentId.get(parent.id);
+    spansByParentId.delete(parent.id);
+
+    // TODO: do we need to sort?
```

Comment:
I wonder if we gain anything by sorting by time. Would we ensure something for this middleware or would it make stuff too brittle for changes in apollo 🤔